operate/schema/src/main/java/io/camunda/operate/store/elasticsearch/NewElasticsearchBatchRequest.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.operate.store.elasticsearch;
+
+import static io.camunda.operate.util.ElasticsearchUtil.UPDATE_RETRY_COUNT;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch._types.Refresh;
+import co.elastic.clients.elasticsearch._types.Script;
+import co.elastic.clients.elasticsearch.core.BulkRequest;
+import co.elastic.clients.elasticsearch.core.BulkResponse;
+import co.elastic.clients.elasticsearch.core.bulk.BulkResponseItem;
+import co.elastic.clients.elasticsearch.core.bulk.UpdateOperation;
+import co.elastic.clients.json.JsonData;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.operate.entities.OperateEntity;
+import io.camunda.operate.exceptions.PersistenceException;
+import io.camunda.operate.store.BatchRequest;
+import jakarta.json.JsonValue;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Batch request that uses new style ElasticsearchClient instead of deprecated RestHighLevelClient.
+ */
+public class NewElasticsearchBatchRequest implements BatchRequest {
+
+  public static final String DEFAULT_SCRIPT_LANG = "painless";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NewElasticsearchBatchRequest.class);
+
+  private final BulkRequest.Builder bulkRequestBuilder = new BulkRequest.Builder();
```

Comment:
I would suggest we pass this in as a parameter to the constructor. This will allow us to write unit tests validating all the various calls (operations(), refresh(), etc) and follows the dependency injection pattern we are using for other dependencies like client and objectmapper